Add `MINUS` keyword support for `Spark` and related dialects

`MINUS` is a supported keyword in Spark and Databricks, but fails in sqlglot:


```python
from sqlglot import parse as parse_sql

def test_parses_minus_statement(migration_index):
    query = "SELECT * FROM old.things MINUS SELECT * FROM old.stuff"
    statements = parse_sql(query, read='databricks')
```

Throws:
```
        if self.error_level == ErrorLevel.IMMEDIATE:
>           raise error
E           sqlglot.errors.ParseError: Invalid expression / Unexpected token. Line 1, Col: 37.
E             SELECT * FROM old.things MINUS SELECT * FROM old.stuff
```

--- a/sqlglot/dialects/spark.py
+++ b/sqlglot/dialects/spark.py
@@ -7,6 +7,7 @@ from sqlglot.dialects.dialect import rename_func, unit_to_var
 from sqlglot.dialects.hive import _build_with_ignore_nulls
 from sqlglot.dialects.spark2 import Spark2, temporary_storage_provider, _build_as_cast
 from sqlglot.helper import ensure_list, seq_get
+from sqlglot.tokens import TokenType
 from sqlglot.transforms import (
     ctas_with_tmp_tables_to_create_tmp_view,
     remove_unique_constraints,
@@ -100,6 +101,13 @@ class Spark(Spark2):
             for q in t.cast(t.List[str], Spark2.Tokenizer.QUOTES)
             for prefix in ("r", "R")
         ]
+
+        KEYWORDS = {
+            **Spark2.Tokenizer.KEYWORDS,
+            # as per https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-setops.html
+            # MINUS is an alias for EXCEPT
+            "MINUS": TokenType.EXCEPT,
+        }
 
     class Parser(Spark2.Parser):
         FUNCTIONS = {

--- a/tests/dialects/test_spark.py
+++ b/tests/dialects/test_spark.py
@@ -237,7 +237,7 @@ TBLPROPERTIES (
         self.validate_all(
             "SELECT * FROM db.table1 MINUS SELECT * FROM db.table2",
             write={
-                "spark": "SELECT * FROM db.table1 EXCEPT SELECT * FROM db.table1",
+                "spark": "SELECT * FROM db.table1 EXCEPT SELECT * FROM db.table2",
             }
         )
 

--- a/tests/dialects/test_spark.py
+++ b/tests/dialects/test_spark.py
@@ -233,6 +233,14 @@ TBLPROPERTIES (
             },
         )
 
+    def test_minus(self):
+        self.validate_all(
+            "SELECT * FROM db.table1 MINUS SELECT * FROM db.table2",
+            write={
+                "spark": "SELECT * FROM db.table1 EXCEPT SELECT * FROM db.table1",
+            }
+        )
+
     def test_spark(self):
         self.assertEqual(
             parse_one("REFRESH TABLE t", read="spark").assert_is(exp.Refresh).sql(dialect="spark"),

--- a/tests/dialects/test_spark.py
+++ b/tests/dialects/test_spark.py
@@ -238,7 +238,7 @@ TBLPROPERTIES (
             "SELECT * FROM db.table1 MINUS SELECT * FROM db.table2",
             write={
                 "spark": "SELECT * FROM db.table1 EXCEPT SELECT * FROM db.table2",
-            }
+            },
         )
 
     def test_spark(self):

--- a/tests/dialects/test_spark.py
+++ b/tests/dialects/test_spark.py
@@ -233,14 +233,6 @@ TBLPROPERTIES (
             },
         )
 
-    def test_minus(self):
-        self.validate_all(
-            "SELECT * FROM db.table1 MINUS SELECT * FROM db.table2",
-            write={
-                "spark": "SELECT * FROM db.table1 EXCEPT SELECT * FROM db.table2",
-            },
-        )
-
     def test_spark(self):
         self.assertEqual(
             parse_one("REFRESH TABLE t", read="spark").assert_is(exp.Refresh).sql(dialect="spark"),
@@ -582,7 +574,8 @@ TBLPROPERTIES (
         )
 
         self.validate_all(
-            "CAST(x AS TIMESTAMP)", read={"trino": "CAST(x AS TIMESTAMP(6) WITH TIME ZONE)"}
+            "CAST(x AS TIMESTAMP)",
+            read={"trino": "CAST(x AS TIMESTAMP(6) WITH TIME ZONE)"},
         )
         self.validate_all(
             "SELECT DATE_ADD(my_date_column, 1)",
@@ -830,8 +823,22 @@ TBLPROPERTIES (
                     self.assertEqual(query.sql(name), without_modifiers)
 
     def test_schema_binding_options(self):
-        for schema_binding in ("BINDING", "COMPENSATION", "TYPE EVOLUTION", "EVOLUTION"):
+        for schema_binding in (
+            "BINDING",
+            "COMPENSATION",
+            "TYPE EVOLUTION",
+            "EVOLUTION",
+        ):
             with self.subTest(f"Test roundtrip of VIEW schema binding {schema_binding}"):
                 self.validate_identity(
                     f"CREATE VIEW emp_v WITH SCHEMA {schema_binding} AS SELECT * FROM emp"
                 )
+
+    def test_minus(self):
+        self.validate_all(
+            "SELECT * FROM db.table1 MINUS SELECT * FROM db.table2",
+            write={
+                "spark": "SELECT * FROM db.table1 EXCEPT SELECT * FROM db.table2",
+                "databricks": "SELECT * FROM db.table1 EXCEPT SELECT * FROM db.table2",
+            },
+        )

